Implement Retry Mechanism for HttpExtractor

georg-schwarz opened the pull request:
Closes #403.

## Example
```
block CarsExtractor oftype HttpExtractor {
		url: "https://gist.githubusercontent.com/noamross/e5dasdasd..."; // invalid URL
		retries: 3;
		retryBackoff: 10000;
	}

```
![image](https://github.com/jvalue/jayvee/assets/28054628/4b4a28af-04f3-4668-97bf-19a5d9d79050)

rhazn commented on libs/extensions/std/exec/src/http-extractor-executor.ts:
```diff
@@ -43,14 +46,39 @@ export class HttpExtractorExecutor extends AbstractBlockExecutor<
+      PrimitiveValuetypes.Integer,
+    );
+    const retryBackoff = context.getPropertyValue(
+      'retryBackoff',
```
I am a big fan in including expected units in variable names that are not obvious. Id suggest `retryBackoffMS` or better `retryBackoffMicroseconds`.

georg-schwarz replied:
With the comment about the exponential backoff strategy, does it make sense to remove this prop completely?

rhazn replied:
We could also go with just exponential backoffs, 1, 2, 4 etc seconds. But might as well keep both configurable, right? Might be useful if we add a prop to change backoff strategy in the future.

rhazn reviewed (changes requested):
I think our default strategy should be exponential backoffs and we should enforce some form of minimal delay. Right now it is trivial to missconfigure (or deliberately configure in the case of JV cloud) this to spam a source with 1ms delayed requests.

I'd go with exponential backoff and minimal delay of 2s. So we'd have 2, 4, 8... etc backoffs.

georg-schwarz commented:
- Added a prop `retryBackoffStrategy` with allowed values `exponential` (default) and `linear`.
- Renamed `retryBackoff` to `retryBackoffMilliseconds`
- Prop `retryBackoffMilliseconds` must be 1000 or larger

rhazn reviewed (approved):
It is wrong but intuitively, I'd assume exp backoff to be 1, 2, 4... for 1000ms. Maybe we force a min value of 2000 or do the 1,2,4 by hand?

I am fine with merging it like this as well though.
